---
title: "Figure 2 and supplemental 1-2"
author: "Ruben Schep"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
output:
  html_document:
    theme: journal #cerulean
    highlight: monochrome
    toc: true
    toc_float: true
    code_folding: show
  editor_options:
    chunk_output_type: console
---

# knitr document van Steensel lab
```{r}
StartTime <-Sys.time()
Date <- substr(gsub("-","",Sys.time()),1,8)
library(knitr)

## Select outdir
out.dir = paste0("figures/rs", Date, "/")
dir.create(out.dir)
opts_chunk$set(dev=c('png', 'pdf'), fig.path = file.path(out.dir))
pdf.options(useDingbats = FALSE)
```

# Introduction

## Description of Data

# Data processing

## Path, Libraries, Parameters and Useful Functions

```{r setup, message=FALSE, warnings=FALSE}
# 6-digit Date tag:
Date <- substr(gsub("-","",Sys.time()),3,8)

# libraries:
library(ggplot2)
library(ggbeeswarm)
library(ggpubr)
library(ggrastr)
library(reshape)
library(reshape2)
library(data.table)
library(tidyverse)
library(dplyr)
library(GGally)
library(ggpmisc)
library(pheatmap)
library(RColorBrewer)
library(broom)
library(ggh4x)
library(magrittr)
library(report)


in.dir.date = 20210311
in.dir = paste0("/DATA/projects/DSBrepair/data/R/rs", in.dir.date, "/")


# Color codes
colore <- c("wt" = "#808184", "other" = "black", "NHEJ" = "#e1251b", "MMEJ" = "#26419a", "SSTR" = "#007A4C")
colores <- c("wt" = "#808184", "other" = "black", "NHEJ" = "#e1251b", "MMEJ" = "#26419a", "SSTR" = "#007A4C")
colori  <- c("other_indels" = "black", "NHEJ" = "#e1251b", "MMEJ" = "#26419a", "SSTR" = "#007A4C") 
colora <- c("0" = "#808184", "1" = "#e1251b", "-7" = "#26419a", "ssODN" = "#007A4C") 

DNA_color = c('MMEJ'="#414691", 'ins'="#d3422e", 'wt'="#000000", 'broken'='grey50',
              'nonMMEJ_del'="#8BA6A5", 'SSTR'="#A9AB2E", 'mut'='#7BA68E',
              'active_MMEJ'='#414691', 'mut_active'='#414691', 'cas9'='#A9AB2E')

k_color = c('kc_cut'="#000000", 'kr_perfect'="#BBABB1", 'kr_ins'="#d3422e",
            'kr_MMEJ_min'="#414691", 'kr_MMEJ_max'="#414691", 'kr_MMEJ'="#414691",
            'kr_nonMMEJ'="#8BA6A5", 'kr_SSTR'="#A9AB2E", 'kr_mut'='#7BA68E')

condition_color = c('DMSO'='#000000', 'M3814'='#d0d0d0')
ssODN_color = c('FALSE'='#000000', 'TRUE'='#A9AB2E')

# Clone color code
col_all_7_clones = values = c("RSTP2_1" = '#014040', 
                              "RSTP2_3" = '#02735E',
                              "RSTP2_5" = '#03A678',
                              "RSTP2_9" = '#ffad07',
                              "RSTP2_12" = '#f27005',
                              "RSTP2_15" = '#ce3800',
                              "RSTP2_16" = '#731702')

col_set_6_clones = values = c("RSTP2_1" = '#026150', 
                              "RSTP2_5" = '#039851',
                              "RSTP2_9" = '#04CC50',
                              "RSTP2_12" = '#FF9B07',
                              "RSTP2_15" = '#CA6C04',
                              "RSTP2_16" = '#9A3900')

## Chromatin groups
group_colors=c('late_replicating-LAD-H3K9me2'='#4477AA',
               "H3K9me2"='#6D92A2',
               "late_replicating-LAD-H3K9me2-H3K27me3"='#6D92A2',
               "LAD-H3K9me2-H3K27me3"='#6D92A2',
               "LAD-H3K9me2"='#6D92A2',
               "late_replicating"='#6D92A2',
               "late_replicating-H3K9me2"='#6D92A2',
               "H3K9me2-H3K27me3"='#6D92A2',
               'H3K27me3'='#66CCEE',
               'transcription'='#228833',
               'euchromatin-transcription'='#CCBB44',
               "transcription-late_replicating-LAD-H3K9me2"="#37868A",
               "euchromatin-H3K27me3"="#74555F", 
               "euchromatin-H3K9me2"="#74555F",
               "euchromatin-transcription-late_replicating"="#74555F",
               'euchromatin'='#EE6677',
               "transcription-LAD-H3K9me2"="#37868A",
               'unmarked'='#BBBBBB')

main_groups = c('euchromatin', 'euchromatin-transcription', 'transcription',
                'unmarked', 'H3K27me3', 'late_replicating-LAD-H3K9me2')
main_group_colors=group_colors[main_groups]

color_gradient=c("#f94144", "#f3722c", "#f8961e", "#f9c74f", "#90be6d",
                 "#43aa8b", "#577590")
# Smaller chromatin group colors
group2_colors=c('euchromatin'='#EE6677',
                'transcription'='#228833',
                "mixed_chromatin"="#74555F",
                'unmarked'='#BBBBBB',
                "other"='#BBBBBB',
                'H3K27me3'='#66CCEE',
                "other_heterochromatin"='#6D92A2',
                'triple_heterochromatin'='#4477AA')

main_groups2 = c("euchromatin", 'transcription', 'unmarked',
                 'H3K27me3', "triple_heterochromatin")
main_group2_colors=group2_colors[main_groups2]


main_groups3 = c("euchromatin", 'transcription', 'other',
                 'H3K27me3', "triple_heterochromatin")
main_group3_colors=group2_colors[main_groups3]

main_groups = c('euchromatin', 'transcription',
                'unmarked', 'H3K27me3', 'late_replicating-LAD-H3K9me2')
main_group_colors=group_colors[main_groups]

feature_fill = c(insulator="#a6b5a3", repressive="#304441", euchromatin="#cda4cc",
                 transcribing="#FBB040", HDAC="#aac9e0", accessibility='#cebc85',
                 methylation='#7dc98f')

col_group_clones = list(clone = col_set_6_clones,
                        domain = group2_colors, 
                        group = feature_fill)

```

## Custom functions
Functions used thoughout this script.
```{r, message=FALSE, warnings=FALSE}
changeLevels <- function(object, column = "chr_r", levels = c(paste0("chr", 1:22), "chrX")) {
  object[, column] <- as.character(object[, column])
  object[, column] <- factor(object[, column], levels)
  object
}
```

# Data import

### Chromosome and chromatin data
```{r chromosome data and mapping, message=FALSE, warnings=FALSE}
map_clones = read_table("/DATA/projects/DSBrepair/data/cl20220321_cured_mapping_clones.txt", col_names = c("chr", "start", "barcode", "clone")) %>%
  filter(chr != "chrY") %>%
  mutate(chr = factor(chr, levels = c("chr1", "chr2", "chr3", "chr4", "chr5", 
                                      "chr6", "chr7", "chr8", "chr9", "chr10", 
                                      "chr11", "chr12", "chr13", "chr14", "chr15", 
                                      "chr16", "chr17", "chr18", "chr19", "chr20", 
                                      "chr21", "chr22", "chrX")),
         start = start / 1e6)

chromsizes = read_table("/DATA/projects/DSBrepair/data/rs20220602_chromsizes_hg38_UCSC.txt", col_names = T) %>% 
  filter(chr != "chrY") %>%
  mutate(chr = factor(chr, levels = c("chr1", "chr2", "chr3", "chr4", "chr5", 
                                      "chr6", "chr7", "chr8", "chr9", "chr10", 
                                      "chr11", "chr12", "chr13", "chr14", "chr15", 
                                      "chr16", "chr17", "chr18", "chr19", "chr20", 
                                      "chr21", "chr22", "chrX")),
         start = start / 1e6)

centromeres = read_table("/DATA/projects/DSBrepair/git/EpiScreen/rs20220714_UCSC_centromeres_24501022.txt") %>%
  group_by(chr) %>%
  filter(chr != "chrY") %>%
  mutate(chr = factor(chr, levels = c("chr1", "chr2", "chr3", "chr4", "chr5", 
                                      "chr6", "chr7", "chr8", "chr9", "chr10", 
                                      "chr11", "chr12", "chr13", "chr14", "chr15", 
                                      "chr16", "chr17", "chr18", "chr19", "chr20", 
                                      "chr21", "chr22", "chrX")),
         start = start / 1e6,
         end = end / 1e6) %>%
  summarise(start = min(start), # earliest start of centromere
            end = max(end)) # furthest end of centromere

setnames(centromeres, old = c("start", "end"), new = c("cen_start", "cen_end"))

chrom_sizes = chromsizes %>% left_join(centromeres)
```

### Chromatin data
```{r chromatin data, message=FALSE, warnings=FALSE}
# Set Clone list
clone_dt = fread('/DATA/projects/DSBrepair/data/cl20220321_cured_mapping_clones.txt',
                 col.names=c('seqnames', 'pos', 'barcode', 'clone'))

setkey(clone_dt, 'barcode')

chip_dt = fread('/DATA/projects/DSBrepair/results/R/cl20220322_zscores_chip_clone_set.txt') %>% 
  filter(pool == "clone_set")  %>%
  select(-pool, -binsize) 

setkey(chip_dt, 'ID')

# Load Lmnb1 DamID data
LMNB1_dt = fread('/DATA/projects/DSBrepair/data/tracks_20220321_cured/dam/clone_set-2000_LMNB1.txt',
                 key='name')
setnames(LMNB1_dt, c("name", "z_score"), c("ID", "LMNB1"))
LMNB1_dt = LMNB1_dt[, c("ID" , "LMNB1")]

chip_LMNB1_dt = merge(chip_dt, LMNB1_dt)

# Read domain data for all the clones
domain_dt = readRDS('/DATA/projects/DSBrepair/data/R/cl20220322_clones_chip_domains.RDS')

domain_dt[,group2:=group]
domain_dt[group=='euchromatin-transcription', group2:='euchromatin']

domain_dt[grepl('transcription', group2) & group!='transcription',
          group2:='mixed_chromatin']

domain_dt[grepl('euchromatin', group2) & group2!='euchromatin',
          group2:='mixed_chromatin']

domain_dt[group%in%c('late_replicating-LAD-H3K9me2',
                     'late_replicating-LAD-H3K9me2-H3K27me3'),
          group2:='triple_heterochromatin']

domain_dt[grep('H3K27me3', group2), group2:='H3K27me3']
domain_dt[!group2%in%names(group2_colors), group2:='other_heterochromatin'] 
```


## simple dt and loesss
```{r}
simple_dt = fread('/DATA/projects/DSBrepair/data/R/cl20220622_timeseries_pools_73set_simple.txt')
```


### Indel data
```{r load indel data, message=FALSE, warnings=FALSE}
pathway_dt = fread('/DATA/projects/DSBrepair/data/R/cl20220622_timeseries_pools_59set.txt')
pathway_dt[, used:=ifelse(time<50, 1, 0.3)]
pathway_dt[, group:=domain_dt[barcode,'group']]
pathway_dt[is.na(group), group:='unmarked']

endseq_scaled = fread(paste0('/DATA/projects/DSBrepair/data/R/',
                             'cl20220530_scaled_endseq_ratios.tsv'),
                      sep='\t')
endseq_scaled[,group:=domain_dt[barcode, 'group']]
endseq_scaled[, pathway:='broken']
endseq_dt = fread(paste0('/DATA/projects/DSBrepair/data/R/',
                         'cl20220517_corrected_broken_fraction_endseq.txt'))
endseq_dt[, pathway:='broken']
endseq_dt[, ssODN:=F]

endseq_clone5 = fread(paste0('/DATA/projects/DSBrepair/data/R/',
                             'cl20220525_endseq_clone5_12h.txt'))

rel_bc_count = pathway_dt[, list(count=sum(count)),
                          by=c('barcode', 'clone', 'bio', 'tech', 'condition',
                               'time', 'ssODN', 'used')]

rel_bc_count[, rel:=count/sum(count),
             by=c('clone', 'bio', 'tech', 'condition', 'time', 'ssODN')]

rel_bc_count[, rel_broken:=rel/rel[time==0],
             by=c('barcode', 'clone', 'bio', 'tech', 'condition', 'ssODN')]


pathway_dt[, mean_diff := ratio-mean(ratio),
           by=c('barcode', 'clone', 'pool', 'bio', 'condition', 'ssODN',
                'group', 'pathway', 'time')]

rel_bc_count[, mean_diff := rel_broken-mean(rel_broken),
             by=c('barcode', 'clone', 'bio', 'condition', 'ssODN', 'time')]


pathway_sd = pathway_dt[time <= 48 & condition=='M3814',
                        list('sd_pathway'=sd(mean_diff)),
                        by='clone']
relative_sd = rel_bc_count[time <= 48 & condition=='M3814',
                           list('sd_relative'=sd(mean_diff)),
                           by='clone']

pathway_sd = pathway_dt[time <= 48 & condition=='M3814' & ssODN==T,
                        list('sd_pathway'=sd(mean_diff)),
                        by='barcode']
relative_sd = rel_bc_count[time <= 48 & condition=='M3814' & ssODN==T,
                           list('sd_relative'=sd(mean_diff)),
                           by='barcode']
```

### Kinetic Rates
```{r}
path = "/DATA/projects/DSBrepair/data/python/"

singlebc_dt = fread(paste0(path, 'cl20220627_timeseries_fit_endseq_1es_max5_73set_endseq_conditions.txt'))
singlebc_dt[, start:='single_bc']


rates_singlebc_dt = fread(paste0(path, 'cl20220627_rate_constants_endseq_1es_73set_endseq_conditions.txt'))
rates_singlebc_dt[, start:='single_bc']


step1of3_dt = fread(paste0(path, 'cl20220622_timeseries_fit_step1of4_0.24_48h_59set.txt'))
step1of3_dt[, start:='step1of3_59set']


rates_step1of3_dt = fread(paste0(path, 'cl20220622_rate_constants_step1of4_0.24_48h_59set.txt'))
rates_step1of3_dt[, start:='step1of3_59set']


step1of2_dt = fread(paste0(path, 'cl20220622_timeseries_fit_step1of2_rel48_59set.txt'))
step1of2_dt[, start:='step1of2_59set']


rates_step1of2_dt = fread(paste0(path, 'cl20220622_rate_constants_step1of2_rel48_59set.txt'))
rates_step1of2_dt[, start:='step1of2_59set']

step2of2_59_dt = fread(paste0(path, 'cl20220623_timeseries_fit_step2of2_relative48_59set.txt'))
step2of2_59_dt[, start:='step2of2_59set']

rates_step2of2_59_dt = fread(paste0(path, 'cl20220623_rate_constants_step2of2_relative48_59set.txt'))
rates_step2of2_59_dt[, start:='step2of2_59set']

step2of3_59_dt = fread(paste0(path, 'cl20220623_timeseries_fit_step2of3_rel48_59set.txt'))
step2of3_59_dt[, start:='step2of3_59set']

rates_step2of3_59_dt = fread(paste0(path, 'cl20220623_rate_constants_step2of3_rel48_59set.txt'))
rates_step2of3_59_dt[, start:='step2of3_59set']


step3of3_59_dt = fread(paste0(path, 'cl20220627_timeseries_fit_step3of3_rel48_59set.txt'))
step3of3_59_dt[, start:='step3of3_59set']

rates_step3of3_59_dt = fread(paste0(path, 'cl20220627_rate_constants_step3of3_rel48_59set.txt'))
rates_step3of3_59_dt[, start:='step3of3_59set']

step3of3_59_SSTR_dt = fread(paste0(path, 'cl20220627_timeseries_fit_step3of3_rel48_59set_SSTRperfect.txt'))
step3of3_59_SSTR_dt[, start:='step3of3_59set_SSTRperfect']

rates_step3of3_59_SSTR_dt = fread(paste0(path, 'cl20220627_rate_constants_step3of3_rel48_59set_SSTRperfect.txt'))
rates_step3of3_59_SSTR_dt[, start:='step3of3_59set_SSTRperfect']

# cl20220613_timeseries_fit_clones_step3of4_M3814_73set_48h.txt
# cl20220613_rate_constants_clones_step3of4_M3814_73set_48h.txt

full_dt = rbind(step2of3_59_dt[ssODN==T & clone!='RSTP2_3', ],
                step1of2_dt[clone!='RSTP2_3', ],
                step3of3_59_dt[clone!='RSTP2_3', ])
rates_full_dt = rbind(rates_step2of3_59_dt[ssODN==T & clone!='RSTP2_3', ],
                      rates_step1of2_dt[clone!='RSTP2_3', ],
                      rates_step3of3_59_dt[clone!='RSTP2_3', ])
full_dt[, step:=ifelse(start=='step1of2_59set', '1step', '3step')]
rates_full_dt[, step:=ifelse(start=='step1of2_59set', '1step', '3step')]

# full_dt = rbind(bflow_dt, ssODN2c_dt, ssODN1e_dt, ssODN2b_dt)
# rates_full_dt = rbind(rates_bflow_dt, rates_ssODN2c_dt, rates_ssODN2a_dt,
#                       rates_ssODN2b_dt)
#
# full_dt = rbind(ssODN_dt, SSE_dt)
#
# full_dt = rbind(bflow_dt, ssODN_dt, ssODN2a_dt, ssODN2b_dt, step3_dt, step4_dt,
#                 step3b_dt, step4b_dt)

# full_dt = rbind(bflow_dt, ssODN_dt, ssODN2a_dt, ssODN2b_dt, step3_dt, step4_dt)


full_dt[, group:=domain_dt[barcode,'group']]
full_dt[is.na(group), group:='unmarked']


#
# rates_full_dt = rbind(rates_ssODN_dt,
#                       rates_SSE_dt)
#
# rates_full_dt = rbind(rates_bflow_dt, rates_ssODN_dt, rates_ssODN2a_dt,
#                       rates_ssODN2b_dt, rates_step3_dt, rates_step4_dt,
#                       rates_step3b_dt, rates_step4b_dt)
#
# rates_full_dt = rbind(rates_bflow_dt, rates_ssODN_dt, rates_ssODN2a_dt,
#                       rates_ssODN2b_dt, rates_step3_dt, rates_step4_dt)
rates_full_dt[, group:=domain_dt[barcode,'group']]
rates_full_dt[is.na(group), group:='unmarked']
rates_full_dt[, constant:=factor(constant,
                                 levels=c('kc_cut', 'kr_perfect', 'kr_ins',
                                          'kr_MMEJ_min', 'kr_MMEJ_max', 'kr_MMEJ',
                                          'kr_nonMMEJ', 'kr_SSTR'))]

full_dt[, not_broken := 1-broken]
full_dt[, rel_broken := not_broken/sum(not_broken) * length(not_broken),
        by=c('clone', 'pool', 'bio', 'condition', 'ssODN', 'time', 'start')]

full_melt = data.table::melt(full_dt, measure.vars=c('ins', 'MMEJ', 'nonMMEJ_del',
                                                     'broken', 'wt', 'SSTR'),
                             variable.name='pathway', value.name='ratio')

full_melt[, real_ratio:= ifelse(pathway=='broken', ratio, ratio * not_broken),
          by=c('barcode', 'clone', 'pool', 'bio', 'condition', 'ssODN',
               'time', 'start')]
```


## Data processing
```{r preprocess experiments}

```


### Loess fits
```{r}
rel_bc_count = simple_dt[barcode!='ATCTCATCTTATTTCC', list(count=sum(count)),
                          by=c('barcode', 'clone', 'bio', 'tech', 'condition',
                               'time', 'ssODN', 'used')]

rel_bc_count[, rel:=count/sum(count),
             by=c('clone', 'bio', 'tech', 'condition', 'time', 'ssODN')]

rel_bc_count[, rel_broken:=rel/rel[time==0],
             by=c('barcode', 'clone', 'bio', 'tech', 'condition', 'ssODN')]

rel_bc_count[, rel_count:=rel * length(unique(barcode)),
             by=c('clone', 'bio', 'tech', 'condition', 'ssODN')]


rel_bc_med = rel_bc_count[, list(rel_measured = median(rel_broken),
                                 rel_count = median(rel_count)),
                          by=c('barcode', 'clone', 'condition',
                               'ssODN', 'time')]
fit_loess <- function(dt){
    time = 0:48
    fit = loess(rel_measured ~ time, data=dt)
    print(fit)
    p = predict(fit, time)
    return(list(time=time, loess=p))
}


loess_dt = rel_bc_med[time <= 48, fit_loess(.SD),
                        by=c('barcode', 'clone', 'condition', 'ssODN')]
loess_dt[,group:=domain_dt[barcode, 'group']]
setkey(clone_dt, 'barcode')
loess_dt[,seqnames:=clone_dt[barcode, 'seqnames']]
```

# Main Figure 1
## Panel 1A
```{r Fig1A chromosome ideogram, fig.height=6, fig.width=8}

```

### Figure legend
*(a)* Chromosome maps of the mapped IPR locations of both pools that passed the mutagenesis assay cutoff. 

## Panel 1B
```{r Fig1B individual barcodes, fig.height=5, fig.width=7}

```
### Figure Legend

*(c)* 

## Panel 1C
```{r Fig1C indel spectra, fig.height=6, fig.width=10}

```

### Figure legend 

*(b)* 




# Supplementary Figure 1

## Panel S1A
```{r FigS1A chromideograms, fig.height=11, fig.width=4}
# remove clone 3 as it is wonky, in cell culture and in the indel data.
map_clones_no3 = map_clones %>% filter(clone != "RSTP2_3")

p = ggplot() +
  geom_segment(data = chrom_sizes,
               aes(x = chr, xend = chr, y = 0, yend = cen_start),
               lineend = "round", color = "lightgrey", size = 7) +
  geom_segment(data = chrom_sizes,
               aes(x = chr, xend = chr, y = cen_end, yend = start),
               lineend = "round", color = "lightgrey", size = 7) +
  geom_segment(data = map_clones,
               aes(x = as.integer(chr) - 0.3, xend = as.integer(chr) + 0.3,
                   y = start, yend = start, color = clone),
               size = 3) +
  geom_segment(data = chrom_sizes,
               aes(x = as.integer(chr) - 0.2, xend = as.integer(chr) + 0.2,
                   y = cen_start, yend = cen_start),
               size = 2) +
  geom_segment(data = chrom_sizes,
               aes(x = as.integer(chr) - 0.2, xend = as.integer(chr) + 0.2,
                   y = cen_end, yend = cen_end),
               size = 2) +
  theme_classic() +
  scale_color_manual(values = col_set_6_clones) +
  xlab("Chromosome") + 
  ylab("Position [Mb]") +
  coord_flip()

pdf('/DATA/projects/DSBrepair/git/DSB_TRIP_Kinetics/figures/rs20220729_mapping_6_clones.pdf', width=6, height=8)
print(p)
dev.off()
```
### Figure Legend
*(a)* Chromosome maps of the uniquely mapped IPR locations per clone.

## Panel S1B
<!-- start Christ -->
```{r FigS1B clone5 heatmap Christ, fig.height=6, fig.width=8}
clones_barcodes = pathway_dt %>%
  filter(clone != "RSTP2_3") %>%
  pull(barcode) %>% unique()

chip.matrix = chip_LMNB1_dt %>% filter(ID %in% clones_barcodes) %>% column_to_rownames(., var = "ID") %>% t()

clones_domains.df = as.data.frame(domain_dt[, c("barcode", "group2")]) %>% filter(barcode %in% clones_barcodes)
colnames(clones_domains.df) = c("ID", "group")

clones_domains.df = merge(clones_domains.df, bc_clones, by.x="ID", by.y="barcode")

clones_domains.df$IPR = paste0("IPR", seq(1:dim(clones_domains.df)[1]))
rownames(clones_domains.df) = clones_domains.df$IPR

chip_clones = chip_LMNB1_dt[clones_domains.df, ]

clones_domains_annotation = clones_domains.df[, c("group", "clone")]
colnames(clones_domains_annotation) = c("domain", "clone")

clones_domains.df = clones_domains.df[, "group", drop = FALSE]







bc_clones = unique(pathway_dt[ , c("barcode", "clone")]) %>%
  filter(clone != "RSTP2_3")



clustering = data.frame(row.names=c('LMNB1', 
                                    # 'late_replicating', 
                                    'H3K9me2', 'H3K9me3',
                                    'H3K27me3', 'EZH2', 
                                    'CTCF', 'SMC3',
                                    'HDAC3', 'HDAC2', 'HDAC1', 
                                    'H3K4me1', 'H3K4me2', 'H3K4me3', 'H3K27ac',
                                    'H4K5acK8ac', 'H2AFZ', 
                                    #'DNAse', 'Dam', 'm5C',
                                    'H3K79me2', 
                                    #'TTseq', 
                                    'H3K36me3', 'POL2AS2',
                                    'POL2'),
                        group=factor(c(#rep('repressive', 6), 
                          rep('repressive', 5), 
                          rep('insulator', 2),
                          rep('HDAC', 3), 
                          rep('euchromatin', 6),
                          # rep('accessibility', 2), 'methylation',
                          # rep('transcribing', 5),
                          rep('transcribing', 4)),
                          levels=c('transcribing', 'accessibility',
                                   'methylation', 'euchromatin',
                                   'HDAC', 'insulator', 'repressive')))

chom_df = chip_clones %>% column_to_rownames("IPR") %>% select(-group, -ID, -clone) %>% t()

order_vec=rownames(clustering)[nrow(clustering):1]

### Colors
quant_breaks <- quantile(chom_df, c(.01,.99), na.rm = T) # change these numbers according to where you want to set the cutoff
palette_length <- 100
max_quant = max(abs(quant_breaks[1]), abs(quant_breaks[2]))

breaks.chip <- c(seq(-quant_breaks[2], 0, length.out=ceiling(palette_length/2) + 1), 
                 seq(quant_breaks[2]/palette_length, quant_breaks[2], length.out=floor(palette_length/2)))


colors = colorRampPalette(rev(brewer.pal(n = 11, name="RdBu")))(palette_length)

# paletteLength <- 50
# myColor <- colorRampPalette(rev(brewer.pal(n = 11, name="RdBu")))(paletteLength)
# # length(breaks) == length(paletteLength) + 1
# # use floor and ceiling to deal with even/odd length pallettelengths
# myBreaks <- c(seq(min(test), 0, length.out=ceiling(paletteLength/2) + 1), 
#               seq(max(test)/paletteLength, max(test), length.out=floor(paletteLength/2)))

pheatmap(chom_df, color = colors, breaks = breaks.chip, annotation_row=clustering)

pdf('/DATA/projects/DSBrepair/git/DSB_TRIP_Kinetics/figures/rs20220729_heatmap.pdf', width=14, height=8)
pheatmap(chom_df, 
         border_color = F,
         cellwidth = 8,
         cellheight = 8,
         na_col = "white",
         clustering_method = "ward.D2",
         annotation_colors=col_group_clones,
         color = colors, 
         breaks = breaks.chip,
         annotation_row = clustering,
         annotation_col = clones_domains_annotation)
dev.off()

```
<!-- end Christ -->
### Figure Legend
*(B)* Heatmap of the Z-scores of the chromatin features for each IPR in clone5. Z-scores were calculated using the ChIP signals for all the IPRs in the pool. IPRs were clusterd based on the z-scores. IPR numbering is unrelated to figure (2c).  

## Panel S1C
```{r FigS1C all indels, fig.height=5, fig.width=12}

```


```{r}
loessmatrix = loess_dt %>% pivot_wider(values_from = loess, names_from = time) %>% filter(condition == "M3814" & ssODN, clone == "RSTP2_5") %>%
  dplyr::select(-condition, -ssODN, -clone, -seqnames, -group) %>% column_to_rownames(var = "barcode")

rel_bc_count

loessmatrix = rel_bc_count  %>%
  filter(condition == "M3814" & ssODN, clone == "RSTP2_5" & time < 50) %>%
  arrange(time) %>%
  dplyr::select(-condition, -ssODN, -clone, -rel_broken, -rel_count, -count, -used) %>% 
  pivot_wider(values_from = rel, names_from = time) 

for(i in 1:3) {
  for(j in unique(loessmatrix$tech)) {
    mat = loessmatrix %>% 
      filter(bio == 1, tech == j) %>%
      dplyr::select(-bio, -tech) %>%
      column_to_rownames(var = "barcode")
    pheatmap(mat,
             main = paste("Rel counts clone 5 bio", i, "tech", j, sep = " "),
             cluster_cols = F)
  }
}
```


### Figure legend
*(C)* 

## Panel S1D
```{r FigS1D indel spectra seperate pools, fig.height=4, fig.width=12}
pathway_dt[bio=='E1601', bio:=1]
pathway_dt[bio=='E1609', bio:=2]
pathway_dt[bio=='E1626', bio:=3]

sd_dt = pathway_dt[, list(sd=sd(ratio)),
                   by=c('pathway', 'ssODN', 'pool', 'barcode','bio','condition','time')]

sd_dt %<>% replace_na(list(sd = 0)) %>% mutate(filter = ifelse((ssODN == FALSE & pathway == "SSTR"), FALSE, TRUE)) %>% filter(filter == TRUE)


p = ggplot(sd_dt,
           aes(x=time, y=sd, color=pathway)) +
  rasterise(geom_point(size = 0.1, alpha = 0.2), dpi = 300, scale = 1) +
  geom_smooth(se=F, size = 2) +
  geom_vline(xintercept=48) +
  theme_bw() +
  scale_color_manual(values=DNA_color) +
  facet_nested(rows=vars(ssODN, condition))

pdf('/DATA/projects/DSBrepair/scratch/rs20220603_sd_tech_rep_jitter.pdf', useDingbats=F, width=5, height=10)
print(p)
dev.off()
```

### Figure legend
*(D)* 


## Panel S2E
```{r FigS2E correlation between replicates, fig.height=5, fig.width=6}
pathway_pairs = pathway_dt %>% 
  group_by(barcode, time, condition, pathway, bio, ssODN, clone) %>% 
  distinct(ratio, barcode, time, condition, pathway, bio, ssODN, clone) %>%
  summarise(
    av.ratio = sum(ratio)/4) %>% 
  dplyr::distinct() %>%  
  pivot_wider(., names_from = bio, values_from = av.ratio) %>%
  mutate_all(~replace(., is.na(.), 0)) %>%
  unite(condition, condition, ssODN)


pdf('/DATA/projects/DSBrepair/scratch/rs20220602_correlations.pdf', width=10, height=10)
for (i in unique(pathway_pairs$condition)) {
  if(grepl("FALSE", i)) {
    d = pathway_pairs %>% 
      filter(condition == i & time < 50 & pathway != "SSTR")
    p = ggpairs(d, columns = 6:8, ggplot2::aes(colour=pathway), title = i) + 
      scale_fill_manual("legend", values = DNA_color) +
      scale_color_manual("legend", values = DNA_color) +
      theme_bw(base_size = 16)
    print(p)
    # p.rast = rasterize(p, layers='Point', dpi=300)
    # print(p.rast)
    
  } else {
    d.sstr = pathway_pairs %>% 
      filter(condition == i & time < 50)
    p = ggpairs(d.sstr, columns = 6:8, ggplot2::aes(colour=pathway), title = i) + 
      scale_fill_manual("legend", values = DNA_color) +
      scale_color_manual("legend", values = DNA_color)+
      theme_bw(base_size = 16)
    print(p)
    # p.rast = rasterize(p, layers='Point', dpi=300)
    # print(p.rast)
  }
}
dev.off()
```

### Figure legend
*(E)* ) 

# Bibliography
```{r citations}
cite_packages()
```


# Session Info
```{r}
paste("Run time: ",format(Sys.time()-StartTime))
getwd()
date()
sessionInfo()
```
